crates/node_binding/src/lib.rs

```diff
@@ -181,6 +184,13 @@ enum TraceState {
 
 #[ctor]
 fn init() {
+  #[cfg(not(debug_assertions))]
```

Comment:
> Limited to `production builds`, otherwise it's too error-prone.

Is this restriction only available during `rspack build`, not `rspack dev`? Or rspack build debug / release binding?

Reply:
I was wondering the same thing. For example in https://github.com/web-infra-dev/rspack/issues/6785, it's happening with dev as well. If that env applies to production only, does it imply we won't be able to start a dev if we use highlight.js?

Reply:
It’s not related to rspack mode

Reply:
Just confirmed that the `production builds` means the production build of Rspack bindings. So this fix should resolve #6785 in both dev/prod modes.